src/Workspaces/VisualBasic/Portable/CodeGeneration/VisualBasicSyntaxGenerator.vb

```diff
@@ -248,6 +248,10 @@ Namespace Microsoft.CodeAnalysis.VisualBasic.CodeGeneration
         End Function
 
         Public Overrides Function TypeExpression(typeSymbol As ITypeSymbol) As SyntaxNode
+            If typeSymbol.IsSystemVoid Then
```

Comment:
Firstly, it's not technically possible to return `System.Void` from a function. It's a special type representing the absence of a value. The compiler flat out disallows this scenario.

Second, the situation arises when you obtain a symbol from an existing interface and pass that symbol onto `SyntaxGenerator.MethodDeclaration`.  If there are any `void` methods in that interface, VB's `SyntaxGenerator` implementation balks.

Lastly, please note my comments in the PR about altering the approach altogether. I think it would be better if the abstract `SyntaxGenerator` checked if the return type is void and, if so, passes through `null`. Both the C# and VB syntax generators would then operate correctly. I just want someone from MS to verify my thinking before I change the PR.